rfcs/2022-1-10-design-tokens.md

```diff
@@ -0,0 +1,66 @@
+
+3. **Interoperability** - Design tokens allow a huge amount of flexibility for both moving a design system to a new platform, or creating a new design system. Should an iOS/Android app ever be built, a new web application created, etc. that system would have a framework ready to utilize for its theming which would be consistent with the other brand applications.
+
+4. **Opportunity for theming** - Decoupling design rules from the codebase allowing the transposition-ing of these static design decisions in various ways, i.e. theming/high contrast. By replacing all tokens in a system with new tokens you can essentially theme swap, assuming the consuming system is utilizing tokens at all levels appropriately.
+
+## Risks
+
+There are few to no risks involved in implementing a token system. Implementation would be incremental and with 1 to 1 testing of output against the current system.
+
+## Scope/Outline
+
+**An outline for how this process could be rolled out for our current system:**
+
+1. A subset of the total token library along with helpers will be created in Typescript and will live in the CMSWDS repository within the packages folder. The following tasks could be completed over the course of two development sprints. This subset will include:
+
+   - The full set of colors used by each system represented in a typescript module.
+   - Naming convention determined with collaboration with the design team.
+   - Type definitions for colors and color format transform helper functions.
+   - _Theme_ typescript modules defined for Core, Hc.goc and M.gov which define Sass variable outputs that map token values to the current color definitions
+   - A webpack/grunt build script which can be implemented in the current build process toolchain to compile these modules into Sass style variables to be imported into each package.
+   - A set of tests scripts to validate data integrity and compare the variable output to the current variable settings to ensure data parity.
+
+2. Once this concept has been validated and released, work on other token types can progress with: measure, spacing, font, animation and other values, utilizing this toolkit. One set of tokens can be ported at a time, broken up into single-sprint long tasks.
+
+3. A build tool will be developed to export tokens from the token library to Sketch variables which can be imported by designers to ensure they are working with the correct set of values. If the choice to support other design tools ever occurs, a new export process would be architecturally easy to implement for almost any API. This could be done over the course of one development sprint.
```

Comment:
This `Scope` section is great. These are separated into pretty discrete tasks. For this third one, I wonder if
1. This could be part of our MVP, which would move it higher in priority than number 3
2. If we should maybe try to tackle a more basic token like _spacing_ first so we can squeeze something like the sketch integration in sooner and prove out more of the whole system before expanding.
Thoughts?

Reply:
Yes makes more sense for #3 to be in place of #2, i've updated that. 

Regarding the type of token to start with, I think color will have the biggest impact/wow factor that would be a good way to kick things off.  I've got the basics complete now I don't think it will be too much to work to possibly even have much of this done this sprint.  I'll grab some of your time tomorrow or monday so you can see where i'm at.

Reply:
Regarding colors: Okay, cool!